File: leetcode/203.remove-linked-list-elements/ts/removeelements.ts
export class ListNode {
    val: number
    next: ListNode | null
    constructor(val?: number, next?: ListNode | null) {
        this.val = (val === undefined ? 0 : val)
        this.next = (next === undefined ? null : next)
    }
}

// N, N
const removeElements = (head: ListNode | null, val: number): ListNode | null => {
    if (head === null) return null

    head.next = removeElements(head.next, val)

    if (head.val === val) return head.next

    return head
};

export default removeElements

File: leetcode/203.remove-linked-list-elements/go/removeelements.go
package removeelements

type ListNode struct {
	Val  int
	Next *ListNode
}

// N, N
func removeElements(head *ListNode, val int) *ListNode {
	if head == nil {
		return nil
	}

	head.Next = removeElements(head.Next, val)

	if head.Val == val {
		return head.Next
	}

	return head
}
